functionary/ui/views/functions.py

```diff
@@ -0,0 +1,76 @@
+from django.http import HttpResponse, HttpResponseForbidden, HttpResponseRedirect
+from django.shortcuts import render
+from django.views.generic.detail import DetailView
+from django.views.generic.list import ListView
+
+from core.auth import Permission
+from core.models import Environment, Function, Task
+
+from ..forms.forms import FunctionForm
+
+
+class FunctionListView(ListView):
+    model = Function
+
+    def get_queryset(self):
+        """Filters functions not in the environment then sorts based
+        on package name, then function name."""
+
+        env_id = self.request.session["environment_id"]
+        return (
+            super()
+            .get_queryset()
+            .filter(package__environment__id=env_id)
+            .order_by("package__name", "name")
+        )
+
+
+class FunctionDetailView(DetailView):
+    model = Function
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        form: FunctionForm = FunctionForm(self.get_object())
```

Comment:
This is just a curiosity more than anything, what's the driver behind the type declaration here?  It's kind of obvious from the assignment what type this is going to end up being and then there's never a point after this where form is reassigned such that type checking would come into play.  Was there a specific thing you were you going for here?

Reply:
No real purpose, I had turned on type suggestions and clicked on some of them. But it didn't really make my editor view look better, so I removed them for your comments.